snapd-glib/snapd-client.c

```diff
@@ -1007,10 +989,27 @@ gboolean snapd_client_connect_finish(SnapdClient *self, GAsyncResult *result,
   return g_task_propagate_boolean(G_TASK(result), error);
 }
 
+static const gchar *get_default_snapd_socket() {
```

Comment:
This makes an unnecessary connect the first time the client is created. I think it would be better to keep connecting on demand.

The current logic is:
```
client.path = default_path
func set_path (path)
    client.path = path
func connect ()
   connect_socket (client.path)
```

I think the new logic should be:
```
client.path = null
func set_path (path)
    client.path = path
func connect ()
    if client.path != null
        connect_socket (client.path)
    else
        if !connect_socket (abstract_path)
            connect_socket (default_path)
```

Reply:
But currently, `snapd_client_get_socket_path()` returns the socket path that will be used even if it hasn't been set before. Doing that would change the behavior, returning NULL if the user didn't set the path. Of course, we can either change that behavior, or we can add a check to know if we are inside a SNAP (although in that case we won't be able to know if we will use the abstract socket or the old one).

Reply:
If the socket path is null, `snapd_client_get_socket_path` could return the first path that will be attempted.

Reply:
ok.

Reply:
Done.